### PR TITLE
Update Truffle and Solium

### PR DESCRIPTION
**Pull request by vittominacori**
# 🚀 Description

This PR is to force update to Truffle 4.1.13 and Solium 1.1.8.

I solved all warnings regarding `visibility-first` and `max-len`.

There are many warnings regarding missing require message (`error-reason`) that everyone should address.

I'm having a `linebreak-style` issue with [RBACCappedTokenMock](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/mocks/RBACCappedTokenMock.sol).
But I think it is a solium issue so I can't resolve it. [Travis test](https://travis-ci.org/OpenZeppelin/openzeppelin-solidity/jobs/406901988#L2821).

- [x] 📘 I've reviewed the [OpenZeppelin Contributor Guidelines](../blob/master/CONTRIBUTING.md)
- [x] ✅ I've added tests where applicable to test my new functionality.
- [x] 📖 I've made sure that my contracts are well-documented.
- [x] 🎨 I've run the JS/Solidity linters and fixed any issues (`npm run lint:all:fix`).

### CODE REVIEW

File: contracts/access/Whitelist.sol [hunk 1]
```diff
@@ -52,8 +52,8 @@ contract Whitelist is Ownable, RBAC {
    * false if all addresses were already in the whitelist
    */
   function addAddressesToWhitelist(address[] _operators)
-    onlyOwner
-    public
+  public
```
Comment (nventuro) [inline contracts/access/Whitelist.sol]: isn't there an indentation missing here? @elopio you experimented a bit with Solium, any ideas on why it didn't catch this?

File: test/ownership/HasNoEther.test.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ contract('HasNoEther', function (accounts) {
     const ownerFinalBalance = await ethGetBalance(accounts[0]);
     const finalBalance = await ethGetBalance(hasNoEther.address);
     assert.equal(finalBalance, 0);
-    assert.isAbove(ownerFinalBalance, ownerStartBalance);
+    assert.isTrue(ownerFinalBalance.greaterThan(ownerStartBalance));
```
Comment (nventuro) [inline test/ownership/HasNoEther.test.js]: There is a separate issue (#1091) for this, try to only commit relevant changes :)

File: contracts/token/ERC721/ERC721Holder.sol [hunk 3]
```diff
@@ -4,7 +4,14 @@ import "./ERC721Receiver.sol";
...
 
 contract ERC721Holder is ERC721Receiver {
-  function onERC721Received(address, address, uint256, bytes) public returns(bytes4) {
+  function onERC721Received(
+    address,
+    address,
+    uint256,
+    bytes)
+  public
+  returns(bytes4)
```
Comment (nventuro) [inline contracts/token/ERC721/ERC721Holder.sol]: Missing indentation?

**Comment by nventuro** [comment]
Turns out that file had Windows-style line endings, which Solium caught. I changed them to Unix-style (same as the rest of the project).

**Comment by vittominacori** [comment]
@nventuro I merged again from master.
If we don't accept this PR we will have conflicts forever and ever because of people are not using the visibility-first rule.

**Comment by nventuro** [comment]
Agreed, these big PRs are better not left open for too long. Thanks once again @vittominacori!

**Pull request merged**